pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -347,9 +346,46 @@ type fileDescription struct {
 	OpenFlag uint32
 }
 
+func (fd *fileDescription) inode() *Inode {
+	return fd.vfsfd.Dentry().Impl().(*kernfs.Dentry).Inode().(*Inode)
+}
+
+func (fd *fileDescription) statusFlags() uint32 {
+	return fd.vfsfd.StatusFlags()
+}
+
 // Release implements vfs.FileDescriptionImpl.Release.
 func (fd *fileDescription) Release() {}
 
+// Release implements vfs.FileDescriptionImpl.OnClose.
+func (fd *fileDescription) OnClose(ctx context.Context) error {
+	// only release when there is no reference to current fd,
+	// doesn't use vfs.FileDescriptionImpl.Release since need ctx
+	// to make request to FUSE server
+	if fd.vfsfd.Refs() != 0 {
+		return nil
+	}
+
+	in := linux.FUSEReleaseIn{
+		Fh:    fd.Fh,
+		Flags: fd.statusFlags(),
+	}
+	// TODO(gvisor.dev/issue/3245): add logic when we support file lock owner.
+	var opcode linux.FUSEOpcode
+	if fd.inode().Mode().IsDir() {
+		opcode = linux.FUSE_RELEASEDIR
+	} else {
+		opcode = linux.FUSE_RELEASE
+	}
+	conn := fd.inode().fs.conn
+	if !conn.NoOpen {
+		// doesn't return any error is analogous to Lunix's behavior.
+		req, _ := conn.NewRequest(auth.CredentialsFromContext(ctx), uint32(kernel.TaskFromContext(ctx).ThreadID()), fd.inode().Ino(), opcode, &in)
```

Comment:
Since Linux doesn't return any error (this line: https://elixir.bootlin.com/linux/v4.4/source/fs/fuse/file.c#L295), my implementation also ignores error

Reply:
I see. So this function always return nil? It might be a good idea to document this